cities/weather_service.py

```diff
@@ -0,0 +1,24 @@
+import os
+
+import httpx
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+async def fetch_weather(city_name: str):
+    api_key = os.getenv("WEATHER_API_KEY")
+    url = (
+        f"http://api.weatherapi.com/v1/current.json?"
+        f"key={api_key}&q={city_name}&aqi=no"
+
+    )
+    async with httpx.AsyncClient() as client:
+        try:
+            response = await client.get(url)
+            response.raise_for_status()
+            data = response.json()
```

Comment:
The `response.json()` method is asynchronous in httpx. You should use `await response.json()` to properly handle the asynchronous response.